packages/cinterop/src/darwin/kotlin/io/realm/kotlin/internal/interop/RealmInterop.kt

```diff
@@ -961,7 +972,16 @@ actual object RealmInterop {
                     }
                 }
             }
-            //    RLM_TYPE_BINARY,
+            is ByteArray -> {
+                cvalue.type = realm_value_type.RLM_TYPE_BINARY
+                cvalue.binary.apply {
```

Comment:
This logic can be replaced with cvalue.binary = memScope.allocArrayOf(value)

Reply:
Ahhh... It would be nice, but `allocArrayOf` returns a pointer of type `ByteVarOf<Byte>` and we need a `UByteVarOf<UByte>`. Perhaps there is something else in the utils but I haven't found it :(